SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -155,6 +215,13 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
     [self presentViewController:vc animated:true completion:nil];
 }
 
+- (void)handleRefresh {
+    [self.dataSource filterEventsWith:@""];
+    self.searchBar.text = @"";
+    [self.tableView scrollToRowAtIndexPath:[NSIndexPath indexPathForRow:0 inSection:0] atScrollPosition:UITableViewScrollPositionTop animated:YES];
```

Comment:
what if you're exiting a MUNI tunnel and scrolling when the data finishes refreshing? might feel a bit jarring to jump to the top?

Reply:
My guess would be that you'd end up getting a fail delegate message from the data source and the refresh wouldn't actually happen.

Reply:
Yeah this:
```
- (void)didFailToUpdateWithError:(nonnull NSError *)error {
    [self handleError:error];
    [self.tableView.refreshControl endRefreshing];
}
```